File: 997-find-the-town-judge/997-find-the-town-judge.py
class Solution(object):
    def findJudge(self, n, trust):
        """
        :type n: int
        :type trust: List[List[int]]
        :rtype: int
        """
        if len(trust) < n - 1: return -1
        
        indegree, outdegree = [0] * (n+1), [0] * (n+1)
        
        for a, b in trust:
            outdegree[a] += 1
            indegree[b] += 1
        
        for i in range(1, n + 1):
            if indegree[i] == n - 1 and outdegree[i] == 0:
                return i
        return -1

File: 997-find-the-town-judge/997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        if (trust.length < n-1) return -1;
        
        int[] score = new int[n+1];
        
        for (int[] group : trust) {
            score[group[0]]--;
            score[group[1]]++;
        }
        
        for (int i = 1; i <= n; i++) {
            if (score[i] == n-1) return i;
        }
        return -1;
    }
}